flow360/component/simulation/cloud.py

```diff
@@ -3,20 +3,52 @@
 import json
 import time
 from datetime import datetime
+from typing import Annotated, Literal, Optional
+
+import pydantic as pd
 
 from flow360.cloud.rest_api import RestApi
 from flow360.component.case import Case
 from flow360.component.interfaces import DraftInterface, ProjectInterface
 from flow360.component.simulation.simulation_params import SimulationParams
 from flow360.component.simulation.web.asset_base import AssetBase
 from flow360.component.surface_mesh import SurfaceMesh
+from flow360.component.utils import is_valid_uuid
 from flow360.component.volume_mesh import VolumeMesh
 from flow360.exceptions import Flow360WebError
 from flow360.log import log
 
 TIMEOUT_MINUTES = 60
 
 
+def _valid_id_validator(input_id: str):
+    is_valid_uuid(input_id)
+    return input_id
+
+
+IDStringType = Annotated[str, pd.AfterValidator(_valid_id_validator)]
+
+
+class DraftPostModel(pd.BaseModel):
+    """Data model for draft post request"""
+
+    name: Optional[str] = pd.Field(None)
+    project_id: IDStringType = pd.Field(serialization_alias="projectId")
+    source_item_id: IDStringType = pd.Field(serialization_alias="sourceItemId")
+    source_item_type: Literal[
+        "Project", "Folder", "Geometry", "SurfaceMesh", "VolumeMesh", "Case", "Draft"
+    ] = pd.Field(serialization_alias="sourceItemType")
+    solver_version: str = pd.Field(serialization_alias="solverVersion")
+    fork_case: bool = pd.Field(serialization_alias="forkCase")
+
+    @pd.field_validator("name", mode="after")
+    @classmethod
+    def _defautl_draft_name(cls, value):
+        if value is None:
+            return "Client " + datetime.now().strftime("%m-%d %H:%M:%S")
```

Comment:
why `Client` ?

Reply:
this can be done in default_factory

Reply:
Changed to default_factory and replaced Client with Draft to be consistent